src/dbnode/persist/fs/mmap_util.go

```diff
@@ -31,65 +31,67 @@ func validateAndMmap(
 	fdWithDigest digest.FdWithDigestReader,
 	expectedDigest uint32,
 	forceMmapMemory bool,
-) ([]byte, error) {
+	reporterOptions mmap.ReporterOptions,
+) (mmap.Descriptor, error) {
+	fd := fdWithDigest.Fd()
+	stat, err := fd.Stat()
+	if err != nil {
+		return mmap.Descriptor{}, err
+	}
+	numBytes := stat.Size()
+	reporterOptions.Context.Size = numBytes
 
 	if forceMmapMemory {
-		return validateAndMmapMemory(fdWithDigest, expectedDigest)
+		return validateAndMmapMemory(fdWithDigest, expectedDigest, numBytes, reporterOptions)
 	}
 
-	return validateAndMmapFile(fdWithDigest, expectedDigest)
+	return validateAndMmapFile(fdWithDigest, expectedDigest, reporterOptions)
 
 }
 
 func validateAndMmapMemory(
 	fdWithDigest digest.FdWithDigestReader,
 	expectedDigest uint32,
-) ([]byte, error) {
-	fd := fdWithDigest.Fd()
-	stat, err := fd.Stat()
-	if err != nil {
-		return nil, err
-	}
-	numBytes := stat.Size()
+	numBytes int64,
+	reporterOptions mmap.ReporterOptions,
```

Comment:
Same here, might be better to just propagate the `mmap.Reporter` then set the `ReporterOptions` here ourselves?